third_party/ipam/controller_util_node/controller_utils.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/*
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+Modifies:
+- Relocate imports to antrea.io/third_party
+- Cleanup unused imports due to code removal and relocation
+- Remove the following unused funcs to avoid additional code imports:
+	DeletePods()
+	SetPodTerminationReason()
+	MarkPodsNotReady()
+	SwapNodeControllerTaint()
+	AddOrUpdateLabelsOnNode()
+*/
+
+package node
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+
+	"k8s.io/api/core/v1"
+
+	"k8s.io/klog/v2"
+)
+
+// RecordNodeEvent records a event related to a node.
+func RecordNodeEvent(recorder record.EventRecorder, nodeName, nodeUID, eventtype, reason, event string) {
```

Comment:
I don't find it's called anywhere.
And even it's called, it will fail if we don't add event create permission to antrea-controller.

Reply:
That's a good point, but since this module has been brought in from another repo, should we strive to keep it unmodified?